### ISSUE TITLE
Default queries after connecting

### ISSUE THREAD

**Issue by max-frai** [blackbeam/rust-mysql-simple#109]
Hi! Some of the libraries emulates mysql protocol for easy api implementation. Such of them is sphinxsearch. Their daemon runs like mysql but it supports limited set of queries and operations.

I found that the only "default" query after connection is getting "max_allowed_packet" value right after handshake (mod.rs:1553). It brokes all the logic and connection fails because sphinxsearch does not know such query.

Is it possible to add something like `no_default_queries : bool` into options. When it set to true no autogenerated queries will be performed from library side. If it's okay for you I can make pull request with changes.

**Comment by blackbeam** [comment]
Hi!
Yes. I'll fix this together with another changes currently in progress (approximately in 2-3 days).

**Comment by max-frai** [comment]
That's not final problem :( 
With simple: `prep_exec("SELECT * FROM bookauthors", ())` I receive error from sphinx daemon: `ERROR 1047 (08S01): unknown command (code=83)`. So I decided to run debug version and look at server logs:
```
receive-QL, cmd=83, buf=26, sock=63, 'SELECT * FROM bookauthors'
post-QL phase=3, buf=39, append=0, write=1, sock=63, tick=835553
pre-QL phase=3, buf=39, append=0, write=1, sock=63, tick=835553
post-QL phase=1, buf=4, append=0, write=0, sock=63, tick=835553
got events=1, tick=869595
 pre-QL phase=1, buf=4, append=0, write=0, sock=64, tick=869595
post-QL phase=2, buf=1, append=0, write=0, sock=64, tick=869595
pre-QL phase=2, buf=1, append=0, write=0, sock=64, tick=869595
receive-QL, cmd=1, buf=1, sock=64, '^A'
epoll remove, ev=0x1, sock=64
cleaned jobs(sock)=1, 0xf12f80(64),
state closing sock=64
```
Your library sends '^A' data 5 times in row after the query. That's why sphinx drops connection and returns error.

I tried to manually connect via mysql tool from console and perform the same query:

```
receive-QL, cmd=3, buf=26, sock=62, '^Cselect * from bookauthors'
QL job started, tick=135456
QL job done, tick=135456
```
And after this it sends result query data back. After fast search in your code I found that you are appending some command constants but I don't know how does mysql protocol works. Maybe you can  help with this?

**Comment by blackbeam** [comment]
Sphinx may not support mysql's binary protocol (`prepare` and `exec` functions). Anyway i'll look into it.

**Comment by blackbeam** [comment]
Looks like [sphinx only supports COM_QUERY](https://github.com/sphinxsearch/sphinx/blob/master/src/searchd.cpp#L16752). This means that your only option is the `Conn::query` method with `Value::into_str` as a way to escape query parameters.

I'll publish new crate version with fixed default queries very soon.

**Comment by max-frai** [comment]
Thank you! For now I temporarily wrote ffi wrapper around c connector of mysql. Waiting for new version.

**Comment by max-frai** [comment]
```
let query = "SELECT * FROM audioauthors WHERE MATCH('Олег Булдаков')  ORDER BY sumdownloads DESC  LIMIT 20;";
Pool::new(options).unwrap().get_conn().unwrap().query(query);
```
This still returns:
```
MySqlError { ERROR 1047 (08S01): unknown command (code=83) }
```

Log from sphinx:
```
receive-QL, cmd=83, buf=111, sock=62, 'SELECT * FROM audioauthors WHERE MATCH('"Олег Булдаков"/1')  ORDER B'
post-QL phase=3, buf=39, append=0, write=1, sock=62, tick=4221325
pre-QL phase=3, buf=39, append=0, write=1, sock=62, tick=4221325
post-QL phase=1, buf=4, append=0, write=0, sock=62, tick=4221325
```

**Comment by blackbeam** [comment]
Just tried following code with latest commit and everything works as expected (with Sphinx 2.2.11):
```rust
let query = "SELECT * FROM test1 WHERE MATCH('my document')  ORDER BY date_added DESC  LIMIT 1;";
let mut conn = Pool::new("mysql://localhost:9306").unwrap().get_conn().unwrap();
for row in conn.query(query).unwrap() {
    println!("ROW: {:?}", row.unwrap());
}
```
Could you please alter your `Cargo.toml` to use git source for the `mysql` dependency and test your code again?
To do this you need to add following to your Cargo.toml:
```toml
mysql = { git = "https://github.com/blackbeam/rust-mysql-simple.git" }
```

**Comment by max-frai** [comment]
Amazing! Works now :) Thank you

**Comment by blackbeam** [comment]
Yay!
I'll publish tomorrow.

**Comment by blackbeam** [comment]
Published as v12.0.0
